interpreter/cling/include/cling/Interpreter/Value.h

```diff
@@ -284,6 +306,19 @@ namespace cling {
     void print(llvm::raw_ostream& Out, bool escape = false) const;
     void dump(bool escape = true) const;
   };
+
+  template <> inline void* Value::getAs() const {
+    if (needsManagedAllocation() || isPointerType())
+      return m_Storage.m_Ptr;
+    return (void*)getAs<uintptr_t>();
+  }
+
+#define X(type, name)                                                   \
+  template <> Value Value::Create(Interpreter& Interp, type val);       \
```

Comment:
Given the resolution of the discussion I think I can now ask again: why `template <>` rather than simple overloads for the cases where we do have an implementation?

Reply:
This kept the similar implementation decision in the past. See [this comment](https://github.com/root-project/root/pull/11252/files#r1085665903) for more information.

> ```c++
> template <typename T> void userCode (T* ptr, T val) { Value::Create<T>(fn->getDouble()) ...}
> ```

Here, in theory `T` might be say an `int` but the value we set can be a double. I'd claim this way is more explicit and maybe will prevent somebody to "optimize out" the alternative `(int) fn->getDouble()` cast. Not sure if that's a real use case but certainly something to consider. 

The other more obvious case is that we can specialize the behavior for pointers or references...

Reply:
> The other more obvious case is that we can specialize the behavior for pointers or references...

How? Through overload? 'Cause partial specialization of function templates isn't a thing.

Reply:
We discussed via voice that should not be an issue - let's go this way and then we can revisit if we need to.